pkg/controller/atlasproject/project_settings.go

```diff
@@ -0,0 +1,99 @@
+package atlasproject
+
+import (
+	"context"
+	"reflect"
+
+	v1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+func ensureProjectSettings(ctx *workflow.Context, projectID string, project *v1.AtlasProject) (result workflow.Result) {
+	if result = syncProjectSettings(ctx, projectID, project); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ProjectSettingsReadyType, result)
+		return result
+	}
+
+	if areProjectSettingsEmpty(project.Spec.Settings) {
```

Comment:
Maybe there is no need to create a separate function for that. Up to you though 😃

Reply:
Yeah, this function was used in multiple places and was supposed to be more complex.
Here just checking for `nil` is fine :)